src/main/java/org/onedatashare/server/module/box/BoxResource.java

```diff
@@ -0,0 +1,379 @@
+package org.onedatashare.server.module.box;
+
+import com.box.sdk.*;
+import com.box.sdk.http.HttpMethod;
+import com.sun.mail.iap.ByteArray;
+import org.apache.commons.io.IOUtils;
+import org.onedatashare.server.model.core.*;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.file.FileSystemException;
+import java.util.*;
+
+public class BoxResource extends Resource<BoxSession, BoxResource> {
+
+    protected BoxResource(BoxSession session, String path, String id) {
+        super(session, path, id);
+    }
+    protected BoxResource(BoxSession session, String path) {
+        super(session, path);
+    }
+
+
+    //tap outgoing
+    //drain incoming
+    public Mono<Stat> stat() {
+
+        return initialize().map(BoxResource::onStat);
+    }
+
+    public Stat onStat() {
+
+        BoxFolder folder = null;
+
+        BoxItem item;
+
+       // folder.getMetadata().getTypeName();
+        //BoxAPIConnection client = new BoxAPIConnection(session.client.getAccessToken());
+        //URL url = new URL();
+        //BoxAPIRequest req = new BoxAPIRequest(session.client, url, "GET");
+        ;
```

Comment:
Extra semi-colon..
You can delete the comments if its not needed.

Reply:
Cleaned up code